packages/time-picker/test/form-input.test.js

```diff
@@ -49,159 +49,341 @@ describe('form input', () => {
+
+    it('should pass validation with a value = max', () => {
+      timePicker.value = '10:00';
+      expect(timePicker.checkValidity()).to.be.true;
+    });
+
+    it('should be invalid when committing a value > max', () => {
       setInputValue(timePicker, '12:00');
-      focusout(comboBox);
-      expect(timePicker.invalid).to.be.equal(false);
+      enter(timePicker.inputElement);
+      expect(timePicker.invalid).to.be.true;
     });
 
-    it('should validate keyboard input (disallowed value)', () => {
-      setInputValue(timePicker, '99:00');
-      expect(timePicker.invalid).to.be.equal(false);
-      focusout(comboBox);
-      expect(timePicker.invalid).to.be.equal(true);
+    it('should be valid when committing a value < max', () => {
+      setInputValue(timePicker, '08:00');
+      enter(timePicker.inputElement);
+      expect(timePicker.invalid).to.be.false;
     });
 
-    it('should fire a validated event on validation success', () => {
-      const validatedSpy = sinon.spy();
-      timePicker.addEventListener('validated', validatedSpy);
-      timePicker.validate();
+    it('should be valid when committing a value = max', () => {
+      setInputValue(timePicker, '10:00');
+      enter(timePicker.inputElement);
+      expect(timePicker.invalid).to.be.false;
+    });
+  });
 
-      expect(validatedSpy.calledOnce).to.be.true;
-      const event = validatedSpy.firstCall.args[0];
-      expect(event.detail.valid).to.be.true;
+  describe('pattern', () => {
+    beforeEach(() => {
+      timePicker = fixtureSync(`<vaadin-time-picker pattern="^1\\d:.*"></vaadin-time-picker>`);
     });
 
-    it('should fire a validated event on validation failure', () => {
-      const validatedSpy = sinon.spy();
-      timePicker.addEventListener('validated', validatedSpy);
-      timePicker.required = true;
-      timePicker.validate();
+    it('should pass validation without value', () => {
+      expect(timePicker.checkValidity()).to.be.true;
+    });
 
-      expect(validatedSpy.calledOnce).to.be.true;
-      const event = validatedSpy.firstCall.args[0];
-      expect(event.detail.valid).to.be.false;
+    it('should fail validation with a value not matching the pattern', () => {
+      timePicker.value = '20:00';
+      expect(timePicker.checkValidity()).to.be.false;
+    });
+
+    it('should pass validation with a value matching the pattern', () => {
+      timePicker.value = '10:00';
+      expect(timePicker.checkValidity()).to.be.true;
+    });
+
+    it('should be invalid when committing a value not matching the pattern', () => {
+      setInputValue(timePicker, '20:00');
+      enter(timePicker.inputElement);
+      expect(timePicker.invalid).to.be.true;
+    });
+
+    it('should be valid when committing a value matching the pattern', () => {
+      setInputValue(timePicker, '10:00');
+      enter(timePicker.inputElement);
+      expect(timePicker.invalid).to.be.false;
+    });
+
+    it('should prevent invalid input', () => {
```

Comment:
**note:** Since `preventInvalidInput` has been deprecated, I guess this test should eventually be modified to test `allowCharPattern` instead, or even dropped.